### PR TITLE
tmpfs: Add support for noatime mount option

### LINKED ISSUE #26102: noatime support for tmpfs mounts
### Feature request description

Add `noatime` support for tmpfs mounts. It's [available in docker](https://docs.docker.com/engine/storage/tmpfs/#options-for---tmpfs) and [may help in reducing I/O](https://serverfault.com/a/1012279).

### PR DESCRIPTION
**Pull request by ArthurWuTW**
### What I did
- Fix #26102

### How I did it
1. Reproduced the Error
```bash
podman run --network host --tmpfs /mytmpfs:noatime alpine sh -c "mount | grep mytmpfs"
```
Output:
```
Error: unknown mount option "noatime": invalid mount option
```

2. Code Trace
Acceptable mount options defined in `mount_opts.go` (`processOptionsInternal`).

3. Unit Test
There is an existing unit test code pattern for mount options in `utils_test.go` (`TestProcessOptions`)
I added a new test in `tests` array.

4. Code Change
add `noatime` option in `mount_opts.go` (`processOptionsInternal`).

5. Documentation
Updated supported mount options to include noatime with a short description.

### How to verify it
1. `make localunit` passes without errors.

2. `make validatepr` without errors.

3. `make podman` and check noatime mount option.

```bash
/home/arthur/podman/bin/podman run --network host --tmpfs /mytmpfs:noatime alpine sh -c "mount | grep mytmpfs"
```
Output: As we can see, `/mytmpfs` has the `noatime` option
```
WARN[0000] Using cgroups-v1 which is deprecated in favor of cgroups-v2 with Podman v5 and will be removed in a future version. Set environment variable `PODMAN_IGNORE_CGROUPSV1_WARNING` to hide this warning. 
tmpfs on /mytmpfs type tmpfs (rw,context="system_u:object_r:container_file_t:s0:c421,c672",nosuid,nodev,noatime,uid=1000,gid=1000)
```

### Release Note
```release-note
Add support for the `noatime` mount option in tmpfs
```

### CODE REVIEW

**Comment by jankaluza** [comment]
I don't see anything wrong with the patch. The tests are missing as @Luap99 pointed out.

File: test/e2e/run_volume_test.go [hunk 1]
```diff
@@ -1122,4 +1122,12 @@ RUN chmod 755 /test1 /test2 /test3`, ALPINE)
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
 	})
+
+	It("podman run --tmpfs with noatime option", func() {
+		session := podmanTest.Podman([]string{"run", "--rm", "--tmpfs", "/mytmpfs:noatime", ALPINE, "grep", "mytmpfs", "/proc/self/mountinfo"})
```
Comment (baude) [inline test/e2e/run_volume_test.go:1127]: consider `podmanTest.PodmanExitCleanly` here ... I think you should also test the negative here too and as a result of the test, ensure the error message you added has been emitted.

Comment (ArthurWuTW) [reply test/e2e/run_volume_test.go:1136]: @baude 
Thank you for reviewing the code and providing your suggestions! 
I'd like to share my findings step by step:

**1. Define Negative Test Case** 
```bash
podman run -v ./testdir:/testdir:noatime alpine
```
I expect an error like
```
the 'noatime' option is only allowed with tmpfs mounts
```
However, the actual error was:
```
Error: invalid option type "noatime"
```
**2. Investigation**
I found this mount error already returned by `ValidateVolumeOpts`, just before `processOptionsInternal`. Also `podman create...` has the same result. 
Untitled Diagram(4)

**3. Expected Error via Rest API**
Interestingly, I tried `podman system serivce...` and it returned the expected error message.
```sh
cd <PATH>/podman/bin
./podman system service --time=0 tcp:127.0.0.1:8080
```
```sh 
# Another terminal
# Create a conatiner
curl -X POST http://127.0.0.1:8080/v4.5.1/libpod/containers/create -H "Content-Type: application/json" -d '{  "name": "myalpine",  "image": "alpine",  "command": ["sleep", "3600"],  "mounts": [{  "type": "bind",  "source": "/home/arthur/podman/bin/testdir",  "destination": "/testdir",  "options": ["noatime"]}]}'

# Response JSON
{"cause":"invalid mount option","message":"the 'noatime' option is only allowed with tmpfs mounts: invalid mount option","response":500}
```

**4. Next Step**
I'd like to add a negative test using curl(Step 3.). Since it needs `podman system service...` running at the beginning, I think I can use BATS...still look around `test/system` directory.. possibly adding a new bats file like `770-create.bats` for noatime test. 
Let me know if this direction makes sense, or if you have any suggestions.

Comment (Luap99) [reply test/e2e/run_volume_test.go:1136]: I think the negative case is that without adding `noatime` then checking that mountinfo should not include the noatime mount option.

> I'd like to add a negative test using curl(Step 3.). Since it needs podman system service... running at the beginning, I think I can use BATS...still look around test/system directory.. possibly adding a new bats file like 770-create.bats for noatime test.

That is not right, test/system runs podman just like the e2e it should not be used for direct API calls generally. If you want api calls look at test/apiv2

**Comment by ArthurWuTW** [comment]
@Luap99 
Thanks for clarifying the negative case.
I was wondering we also need API tests for what @baude suggested. 
> ensure the error message you added has been emitted.

Let me summarize all tests I added for this pull request.

**Unit Test**

- Positive Case: test `func processOptionsInternal` with `noatime` option and `isTmpfs` flag is `true`

- Negative Case: test `func processOptionsInternal` with `noatime` option and `isTmpfs` flag is `false`, error expected

```sh
make localunit
```
```output
--- PASS: TestProcessOptions (0.00s)
...
    --- PASS: TestProcessOptions/noatime_allowed_only_with_tmpfs (0.00s)
...
```

**E2E**

- Positive Case: run with tmpfs mount and `noatime`

- Negative Case: run with tmpfs mount but without `noatime`

```sh
make localintegration FOCUS="podman run --tmpfs with noatime option"
```
```output
...
Ran 1 of 2377 Specs in 26.473 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 2376 Skipped
```

**API**

- Positive Case: tmpfs with `noatime`

- Negative Case: unsupported `noatime` for bind mount, expect new added error message shows up.

```sh
cd test/apiv2
./test-apiv2 44-mounts
```
```output
...
ok 1 [44-mounts] POST containers/create?name=hostconfig_test [-d {"Image":"quay.io/libpod/testimage:20241011","Cmd":["df","-P","/mytmpfs"],"HostConfig":{"Binds":["/tmp/doesnotexist:/test1"],"TmpFs":{"/mytmpfs":"rw,noatime"}}}] : status=201
...
ok 11 [44-mounts] POST libpod/containers/create [-d {"Image":"quay.io/libpod/testimage:20241011","Mounts":[{"type":"bind","source":"/nosuchdir","destination":"/data","options":["noatime"]}]}] : .message ('the 'noatime' option is only allowed with tmpfs mounts: invalid mount option') ~ the 'noatime' option is only allowed with tmpfs mounts
...
```

**Comment by Luap99** [comment]
@giuseppe PTAL, this should be good to merge

**Review by giuseppe — APPROVED** [review]
/lgtm
